mido/midifiles/midifiles.py

```diff
@@ -365,6 +368,13 @@ def length(self):
                              ' for type 2 (asynchronous) file')
 
         return sum(msg.time for msg in self)
+		
+    def msg_count(self):
```

Comment:
I think this may be better defined as overriding the __len__ of the file? https://docs.python.org/2/reference/datamodel.html#object.__len__

This could just be as simple as:
len(my_file)

Maybe more pythonic. If we wanted msg_count, it could just be an alias for len

Reply:
I agree. ``__len__()`` is the way to go here.

It's best to compute this on demand since the message count can change as you add and remove tracks and messages. I propose something like this (off the top of my head):

```
    def __len__(self):
        return sum(len(track) for track in self.tracks)
```

Reply:
I tried that but it returned the length as a metric of time, not messages. I suppose I could just iterate through the messages wherever I import MidiFile, but it seemed weird not to have it as an accessible property.

Reply:
Ah, that's what I was looking for. I didn't want to loop through each time, but I'm relatively new to the python ecosystem still (C++ guy).

Reply:
welcome to python, any questions, never hesitate to reach out!

Reply:
Yeah the __len__ implementation worked well. Strangely the __len__ implementation results in 5 more total messages than the iterative mess I put here for this midi file:
https://github.com/scottmudge/orchestrion/raw/master/data/midis/tf_in_d_minor.mid

But there are also 5 named tracks, so perhaps it's a difference between length (N+1) and total accessible elements (N).

Reply:
I understand your confusion about the `length` attribute. I should probably have chosen a more specific name for it such such as `total_time`.

Reply:
Those 5 messages are `end_of_track` messages that are filtered out when `self.__iter__()` merges all tracks into one for playback.